sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -341,69 +343,160 @@ void initializeHttpRequest(BlockStmt body, HttpRequestContext method) {
     }
 
     private void appendQueryParams(BlockStmt body, HttpRequestContext method) {
-        // Iterate through the query parameters and append them to the url string if they are not null
-        if (!method.getQueryParams().isEmpty()) {
-            // Declare newUrl once
-            Statement newUrlDeclaration = StaticJavaParser.parseStatement("String newUrl;");
-            newUrlDeclaration.setComment(new LineComment("\n Append non-null query parameters"));
-            body.addStatement(newUrlDeclaration);
-            body.tryAddImportToParentCompilationUnit(LinkedHashMap.class);
-            body.addStatement("LinkedHashMap<String, Object> queryParamMap = new LinkedHashMap<>();");
-
-            method.getQueryParams().entrySet().forEach(entry -> {
-                String key = entry.getKey();
-                HttpRequestContext.QueryParameter value = entry.getValue();
-                boolean isValueTypeString = method.getParameters()
-                    .stream()
-                    .anyMatch(parameter -> parameter.getName().equals(value.getValue())
-                        && "String".equals(parameter.getShortTypeName()));
-                if (value.shouldEncode()) {
-                    if (isValueTypeString) {
-                        String encodedKey = "UriEscapers.QUERY_ESCAPER.escape(\"" + key + "\")";
-                        String encodedValue = "UriEscapers.QUERY_ESCAPER.escape(" + value.getValue() + ")";
+        if (method.getQueryParams().isEmpty()) {
+            return;
+        }
+
+        // Add imports and declare variables
+        body.tryAddImportToParentCompilationUnit(LinkedHashMap.class);
+        body.addStatement("LinkedHashMap<String, Object> queryParamMap = new LinkedHashMap<>();");
```

Comment:
Any reason for the value of the map here to be `Object`? Query parameter values should always be string as they're part of the URI

Reply:
It is Object to support values of String or List<String>. This was addressed in this PR here https://github.com/Azure/azure-sdk-for-java/pull/44762#discussion_r2008386044